src/proposed.api.d.ts

```diff
@@ -0,0 +1,131 @@
+     */
+    label: string;
+    /**
+     * A human-readable string which is rendered less prominent in a separate line. Supports rendering of
+     * {@link ThemeIcon theme icons} via the `$(<name>)`-syntax.
+     */
+    detail?: string;
+    /**
+     * A string that should be used when comparing this item
+     * with other items. When `falsy` the {@link JupyterServerCreationItem.label label}
+     * is used.
+     */
+    sortText?: string;
+    /**
+     * Optional flag indicating if this item is picked by default.
+     * If there are no existing servers, and this flag is true, then this item will be picked by default.
+     *
+     * Note: this property is ignored when {@link JupyterServerCollection.createServer createJupyterServer} has been called.
+     */
+    picked?: boolean;
+}
+
+export class JupyterServerCollection extends Disposable {
+    /**
+     * Identifier must be globally unique.
+     */
+    readonly id: string;
+    /**
+     * A human-readable string which is rendered prominent.
+     */
+    label: string;
+    /**
+     * A link to a page providing more information to the user about this item.
+     */
+    documentation?: Uri;
+    /**
+     * Creates an entry in the list of Jupyter Servers from which the user can pick.
+     *
+     * @param {string} id
+     * @param {string} label
+     * @param {GetConnectionInformation} resolveConnectionInformation Gets the connection information for the Jupyter Server.
+     * @return {*}  {JupyterServer}
+     * @memberof JupyterServerCollection
+     */
+    createServer(
+        id: string,
+        label: string,
+        resolveConnectionInformation: () => Promise<JupyterServerConnectionInformation>
+    ): JupyterServer;
+    /**
+     * Creates an entry in the list of Jupyter Servers from which a user can pick to create a server.
+     * Picking an item is expected to result in the eventual creation of a JupyterServer.
+     * I.e. an extension is expected to listen to the `onDidSelect` event and optionally display their own UI and then create a JupyterServer.
+     *
+     * @param {string} label
+     * @param {() => Promise<JupyterServer | undefined>} onDidSelect Callback invoked when this item is selected.
+     * @return {*}  {JupyterServerPicker}
+     * @memberof JupyterServerCollection
+     */
+    createServerCreationItem(
+        label: string,
+        onDidSelect: () => Promise<JupyterServer | undefined>
+    ): JupyterServerCreationItem;
+}
+
+/**
+ * Sample usage extensions.getExtension<JupyterAPI>('ms-ai-tools.jupyter')?.exports;
+ */
+export interface JupyterAPI {
+    /**
+     * Provides the ability to register multiple collections of Jupyter Servers.
+     *
+     * @param {string} id Identifier must be globally unique.
+     * @param {string} label
+     * @return {*}  {JupyterServerCollection}
+     * @memberof JupyterAPI
+     */
+    createServerCollection(id: string, label: string): Promise<JupyterServerCollection>;
```

Comment:
Are collections scoped to the extension? For instance, what happens if two extensions use the same `id`?

Reply:
Yes they are scoped to the extension,
I discussed with Peng the old code requires it to be unique (i.e. if another extension uses the same id, then that could result in problems), but I guess thats a problem that we need to fix.

So, lets go with unique per extension, which is my preference.
However, the challenge here is we need to know the caller (extension), today we get this information by analyzing the stack trace.